src/dropdown/list/DropdownList.tsx

```diff
@@ -0,0 +1,118 @@
+import "./_dropdown-list.scss";
+
+import React, {useRef, useEffect} from "react";
+import classNames from "classnames";
+
+import DropdownListItem, {
+  DropdownOption,
+  TDropdownOptionSelectHandler,
+  TDropdownSelectedOption
+} from "./item/DropdownListItem";
+import {computeScrollAmountToMakeChildVisible} from "../../core/utils/domUtils";
+
+interface DropdownListProps<OptionIdShape extends string> {
+  testid: string;
+  options: DropdownOption<OptionIdShape>[];
+  selectedOption: TDropdownSelectedOption<OptionIdShape>;
+  focusedOption: TDropdownSelectedOption<OptionIdShape>;
+  onSelect: TDropdownOptionSelectHandler<OptionIdShape>;
+  onFocus: TDropdownOptionSelectHandler<OptionIdShape>;
+  onMouseDown?: React.ReactEventHandler<HTMLLIElement>;
+  onMouseUp?: React.ReactEventHandler<HTMLLIElement>;
+  onKeyDown?: TDropdownOptionSelectHandler<OptionIdShape>;
+  role?: "listbox" | "menu" | "combobox";
+  customClassName?: string;
+  ariaLabelledBy?: string;
+  ariaHidden?: boolean;
+  isMultiSelect?: boolean;
+  children?: never;
+  canSelectAlreadySelected?: boolean;
+  noOptionsMessage?: string;
+}
+
+function DropdownList<OptionIdShape extends string>({
+  testid,
+  options,
+  customClassName,
+  role,
+  ariaLabelledBy,
+  ariaHidden,
+  isMultiSelect,
+  selectedOption,
+  focusedOption,
+  onSelect,
+  onFocus,
+  onKeyDown,
+  onMouseDown,
+  onMouseUp,
+  canSelectAlreadySelected,
+  noOptionsMessage
+}: DropdownListProps<OptionIdShape>) {
+  const listRef = useRef<HTMLUListElement>(null);
+  const containerClassName = classNames("dropdown-list", customClassName);
+
+  useEffect(() => {
+    if (listRef.current) {
+      const focusedOptionElement =
+        focusedOption && document.getElementById(focusedOption.id);
+
+      if (focusedOptionElement) {
+        listRef.current.scrollTop += computeScrollAmountToMakeChildVisible(
+          listRef.current,
+          focusedOptionElement
+        );
+      }
+    }
+  }, [focusedOption]);
+
+  useEffect(() => {
+    if (listRef.current) {
+      const selectedOptionElement =
+        selectedOption && document.getElementById(selectedOption.id);
+
+      if (selectedOptionElement) {
+        listRef.current.scrollTop += computeScrollAmountToMakeChildVisible(
+          listRef.current,
+          selectedOptionElement
+        );
+      }
+    }
+  }, [selectedOption]);
+
+  return (
+    <ul
+      ref={listRef}
+      data-testid={testid}
+      role={role}
+      aria-labelledby={ariaLabelledBy}
+      aria-hidden={ariaHidden}
+      aria-multiselectable={isMultiSelect}
+      className={containerClassName}>
+      {options.length ? (
+        options.map(function renderDropdownListItem(option, index) {
+          return (
```

Comment:
I think we can convert this to arrow function to increase readability
```suggestion
        options.map((option, index) => (
```